File: CPP/RomanToInt.cpp
#include <string> 
#include <map>
#include <iostream>

using namespace std;
int romanToInt(string s);
int indice(char c);

int main(){
    cout << romanToInt("III") << endl;
    cout << romanToInt("LVIII") << endl;
    cout << romanToInt("MCMXCIV") << endl;
    return 1;
}

int romanToInt(string s){
    if (s.empty()) {return 0;}
    int acc = 0;
    int prev_val = 0;
    for(int i = s.length() - 1; i >= 0; --i){
        int current_val = indice(s[i]);
        prev_val > current_val ? acc -= current_val : acc += current_val;
        prev_val = current_val;
    }
    return acc;
}

int indice(char c){
    switch (c) {
        case 'I': return 1;
        case 'V': return 5;
        case 'X': return 10;
        case 'L': return 50;
        case 'C': return 100;
        case 'D': return 500;
        case 'M': return 1000;
        default:  return 0;
    }
}

File: Python/RomanToInt.py
def helper(c: str) -> int:
    if c == "I": return 1
    if c == "V": return 5
    if c == "X": return 10
    if c == "L": return 50
    if c == "C": return 100
    if c == "D": return 500
    if c == "M": return 1000
    else: return 0

def romanToInt(s: str) -> int:
    indice = {
        "I" : 1,
        "V" : 5,
        "X" : 10,
        "L" : 50,
        "C" : 100,
        "D" : 500,
        "M" : 1000
    }
    acc = 0
    for i in range(len(s)-1):
       if indice[s[i+1]] > indice[s[i]]: 
          acc -= indice[s[i]]
       else: 
          acc += indice[s[i]]
    acc += indice[s[-1]]
    return acc

def main():
    print(romanToInt("III"))
    print(romanToInt("LVIII"))
    print(romanToInt("MCMXCIV"))
main()